src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,61 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final Random random = new Random();
+    private static final int DEFAULT_LENGTH = 10;
+    private static final String DEFAULT_COLOR = Color.WHITE.toString();
+
+    private int getIndex() {
+        return random.nextInt(4);
+    }
+
+    private int getFirstSide() {
+        return random.nextInt(1,10);
+    }
+
+    private int getSecondSide() {
+        return random.nextInt(1,10);
+    }
+
+    public Figure getRandomFigure() {
+        switch (getIndex()) {
+            case 0:
+                Circle circle = new Circle();
+                circle.setRadius(getFirstSide());
+                circle.setColor(new ColorSupplier().getRandomColor());
```

Comment:
ColorSupplier must be a class field
```suggestion
                circle.setColor(colorSupplier.getRandomColor());
```